src/io/flutter/module/FlutterModuleBuilder.java

```diff
@@ -176,6 +176,10 @@ public boolean validateModuleName(@NotNull String moduleName) throws Configurati
       throw new ConfigurationException("Invalid module name: '" + moduleName + "' - must be a valid Dart identifier.");
     }
 
+    if (!FlutterUtils.isValidPackageName(moduleName)) {
```

Comment:
Maybe do this check first, since it's usually stricter than the previous checks and the error message is clearer. (If the user fixes the mistake based on this message, it will usually pass the others.)